DxDispatch/src/model/JsonParsers.cpp

```diff
@@ -1046,12 +1088,34 @@ Model::BufferDesc ParseModelBufferDesc(const rapidjson::Value& object)
     {
         if (initialValuesField->value.HasMember("value")) 
         {
+            ensureInitialValuesDataType();
             buffer.initialValues = GenerateInitialValuesFromConstant(buffer.initialValuesDataType, initialValuesField->value);
         }
         else if (initialValuesField->value.HasMember("valueStart"))
         {
+            ensureInitialValuesDataType();
             buffer.initialValues = GenerateInitialValuesFromSequence(buffer.initialValuesDataType, initialValuesField->value);
         }
+        else if (initialValuesField->value.HasMember("sourcePath"))
+        {
+            auto [initialValues, initialValuesDataType] = GenerateInitialValuesFromFile(initialValuesField->value);
+
+            // Depending on the file type (.npy vs .dat), the file may have an explict data type.
+            // Use the data type if present, else require initialValuesDataType if not.
+            if (buffer.initialValuesDataType == DML_TENSOR_DATA_TYPE_UNKNOWN)
+            {
+                buffer.initialValuesDataType = initialValuesDataType;
+            }
+            else if (initialValuesDataType != DML_TENSOR_DATA_TYPE_UNKNOWN)
+            {
+                auto fileName = ParseStringField(object, "sourcePath");
+                throw std::invalid_argument(fmt::format("Data type from file '{}' does not match field 'initialValuesDataType'.", fileName));
+            }
+            ensureInitialValuesDataType();
+
+            buffer.initialValues = std::move(initialValues);
+            ensureInitialValuesDataType(); // Raw data requires 'initialValuesDataType'. Typed data already had a type.
```

Comment:
Called twice for some reason.

Reply:
I just wanted to be *extra* sure. (ok, the reason is 1am :D) Also, PR == WIP.